rpc/connection.go

```diff
@@ -169,6 +170,8 @@ type ConnectionTransportTLS struct {
 // Test that ConnectionTransportTLS fully implements the ConnectionTransport interface.
 var _ ConnectionTransport = (*ConnectionTransportTLS)(nil)
 
+const keepAlive = 10 * time.Second
```

Comment:
60 seconds should be enough, shouldn't it? 10 seconds seems like a very aggressive KeepAlive.

Reply:
10s was there before this change, and I don't know how it was chosen. But my guess is it would be the time it takes for use to realize we are disconnected in certain conditions. If my guess is correct, 60s seems a bit long?

Reply:
It doesn't do anything anyway connecting to ELB

Reply:
Hmm I don't understand. Why does it not matter for the connection between client and ELB?

Reply:
http://docs.aws.amazon.com/elasticloadbalancing/latest/classic/config-idle-timeout.html

"Note that TCP keep-alive probes do not prevent the load balancer from terminating the connection because they do not send data in the payload."

Reply:
Although does it not still help detecting disconnections? For example if device goes to sleep, and ELB prunes the connection, and device connects again. If I understand it correctly this makes sure the stale TCP connection gets reset within 10s.